### PR TITLE
status: show only entitlements which exist in the server

### PR DESCRIPTION
**Pull request by renanrodrigo**
If an entitlement does not exist in the Contracts Server, then we don't need to surface it.

## Desired commit type
 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist
 - [x] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

File: uaclient/config.py
Comment (lucasmoura) [inline uaclient/config.py:732]: Why the resources need to be sorted ?

File: uaclient/tests/test_cli_status.py
Comment (lucasmoura) [inline uaclient/tests/test_cli_status.py:19]: Shouldn't we also have a test where we have a service that is not present on the contracts response ?

Comment (renanrodrigo) [reply uaclient/tests/test_cli_status.py:19]: Good point. Will remove one of those

File: uaclient/config.py
Comment (renanrodrigo) [reply uaclient/config.py:732]: We currently rely on it to show them in order when presenting the tabular output

**Review by lucasmoura — APPROVED** [review]
Let's just wait on CI to see if everything is fine

File: uaclient/config.py
Comment (lucasmoura) [reply uaclient/config.py:732]: Sorry, just one more thing. Let's use `get` here to access the resource name. It is unlikely that we will receive a resource without that key, but we could be defensive anyway

**Review by lucasmoura — CHANGES_REQUESTED** [review]

File: uaclient/config.py
Comment (renanrodrigo) [reply uaclient/config.py:732]: Hmm, I would like to keep this as-is, for:
a) we do want to raise an error in this case: a resource without name makes no sense, and no default applies if we don't find one
b) mypy complains that we use it, and we need to do some type juggling to avoid those same errors of a)

**Comment by renanrodrigo** [comment]
I'm checking integration tests

File: uaclient/config.py
Comment (lucasmoura) [reply uaclient/config.py:732]: We could just use `get("name", "")` in that way we would not break if the resource doesn't have a name and we would not break either by trying to access the entitlement on the dict

Comment (renanrodrigo) [reply uaclient/config.py]: oki, done

**Review by lucasmoura — APPROVED** [review]
Let's just wait CI on this one before merging it

**Comment by renanrodrigo** [comment]
According to an elder saying from my people, "Green Jenkins is a sign of good luck" :four_leaf_clover:

**Review by orndorffgrant — APPROVED** [review]
LGTM thanks!

**Pull request merged**
